server-spi-private/src/main/java/org/keycloak/services/clientpolicy/condition/ClientPolicyConditionProvider.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2020 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.keycloak.services.clientpolicy.condition;
+
+import org.keycloak.provider.Provider;
+import org.keycloak.services.clientpolicy.ClientPolicyContext;
+import org.keycloak.services.clientpolicy.ClientPolicyEvent;
+import org.keycloak.services.clientpolicy.ClientPolicyException;
+
+/**
+ * This condition determines to which client a {@link ClientPolicyProvider} is adopted.
+ * The condition can be evaluated on the events defined in {@link ClientPolicyEvent}.
+ * It is sufficient for the implementer of this condition to implement methods in which they are interested
+ * and {@link isEvaluatedOnEvent} method.
+ */
+public interface ClientPolicyConditionProvider extends Provider {
+
+    final String SKIP_EVALUATION = "skip-evaluation";
```

Comment:
Why this exception for notifying skipping the evaluation of the condition is as follows.

Assume that
-  `ClientPolicyConditionProvider.isSatisfiedOnEvent()` returns true if this condition is not evaluated on the event.
- On some event, all conditions of the policy are not evaluated. (e.g. On authorization request, the condition for checking Client App's IP is not possible to be evaluated because the browser is the HTTP Client, not Client App.)
- When this event happens, the executors of the policy are executed because all conditions of the policy return trues even these conditions are not evaluated.

To avoid this situation, I've explicitly distinguish between "evaluated and satisfied (returns true)", "evaluated but not satisfied the condition (returns false)" and "not evaluated (throws exception) ".

Reply:
I don't like the use of an exception in this way.

What about something like:

default Vote applyPolicy(ClientPolicyContext context)

Where Vote is an enum with values YES, NO, ABSTAIN

Reply:
I see, I'll revise codes to  add `public enum ClientPolicyVote` and replace `default boolean isSatisfiedOnEvent(ClientPolicyContext context)` with `default ClientPolicyVoteapplyPolicy(ClientPolicyContext context)` as you suggested.